api/dependencies/auth.py

```
@@ -0,0 +1,142 @@
+            cred = credentials.Certificate("firebase-credentials.json")
+    
+    # Initialize the app
+    firebase_admin.initialize_app(cred)
+    logger.info("Firebase Admin initialized successfully")
+except Exception as e:
+    logger.error(f"Failed to initialize Firebase Admin: {str(e)}")
+
+# Bearer token authentication scheme
+security = HTTPBearer()
+
+async def verify_token(credentials: HTTPAuthorizationCredentials = Depends(security)) -> dict:
```

Ruff (0.8.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)